qiskit_machine_learning/neural_networks/two_layer_qnn.py

```diff
@@ -111,11 +124,11 @@ def __init__(
 
         # construct observable
         self.observable = (
-            observable if observable else PauliSumOp.from_list([("Z" * num_qubits_, 1)])
+            observable if observable is not None else PauliSumOp.from_list([("Z" * num_qubits_, 1)])
         )
 
         # combine all to operator
-        operator = ~StateFn(self.observable) @ StateFn(self._circuit)
+        operator = StateFn(self.observable, is_measurement=True) @ StateFn(self._circuit)
```

Comment:
Does the code do the same? A bit non-obvious replacement.

Reply:
No it doesn't do exactly the same, the new one is correct while the old one isn't 😉 

If you use `~` to indicate a measurement, then the operator will wrongly be complex conjugated. That doesn't happen if you directly use `is_measurement=True`. Since most our observables are hermitian that's usually not a problem but if you have a non-hermitian observable for some reason (e.g. `iX`) then it silently gives wrong results.